README.md

```diff
@@ -29,13 +31,38 @@ cblas.h
 math.h 
 float.h
 stdio.h
+time.h
+string.h
+
+## Formatting convention
+* Allman bracket style
+* 
 
 ## Naming convention
-* Global constants are fully capitilised e.g. R_GRID
+* Global constants are fully capitilised with underscores e.g. R_GRID
+* All global variables are constant and defined at compile time
+* variables and function names should be written in snake case 
+i.e. lower_case_with_underscores
+* variable names should be a noun and should refer to the quantity rather than 
+a Greek letter. e.g. flux instead or psi.
+* function names should begin with a verb e.g. 'get_flux_norm' 
+* boolean variables are simply integers in C. Maybe use short or other inbuilt types
+* Have not used size_t or any other non-inbuilt types
+* Boolean names should begin with is_ prefix to differentiate them from other variables
+* Boolean names should be always be in the "positive" affirmation, with 1 being
+true and 0 being false
+* order of variables should go from largest to smallest e.g. GRID_R_MIN
+as if it was sucessive objects in the OOP paradigme.  
+* Green's functions are prefixed with g_
+
```

Comment:
Nice, I really like it! I will start adopting these things in my code.

Reply:
yeah please note the code doesnt actually abide by most of this stuff yet ... put here more of as a reminder for me